src/web/app/pages-instructor/instructor-course-student-details-page/instructor-course-student-details-page.component.html

```diff
@@ -1,3 +1,128 @@
+                <td class="font-weight-bold">Personal Email</td>
+                <td><span *ngIf="studentProfile.email; else notSpecified">{{studentProfile.email}}</span>
+              </tr>
+              <tr>
+                <td class="font-weight-bold">Institution</td>
+                <td><span *ngIf="studentProfile.institute; else notSpecified">{{studentProfile.institute}}</span>
+              </tr>
+              <tr>
+                <td class="font-weight-bold">Nationality</td>
+                <td><span *ngIf="studentProfile.nationality; else notSpecified">{{studentProfile.nationality}}</span>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+  </div>
+
+  <div class="card border-primary">
+    <div class="card-body fill-plain">
+      <div class="form form-horizontal">
+        <div class="form-group">
+          <div class="col-sm-12">
+            <h4 class="font-weight-bold">Enrollment Details</h4>
+          </div>
+        </div>
+        <div class="form-group student-info-row row">
+          <div class="col-4 text-right">
+            <label>Course</label>
+          </div>
+          <div class="col-8">
+            <a routerLink="/web/instructor/courses/details?courseid={{student.course}}">
+              <span>{{student.course}}</span>
+            </a>
+          </div>
+        </div>
+        <div class="form-group student-info-row row" *ngIf="student.section">
+          <div class="col-4 text-right">
+            <label>Section Name</label>
+          </div>
+          <div class="col-8">
+            <span>{{student.section}}</span>
+          </div>
+        </div>
+        <div class="form-group student-info-row row">
+          <div class="col-4 text-right">
+            <label>Team Name</label>
+          </div>
+          <div class="col-8">
+            <span>{{student.team}}</span>
+          </div>
+        </div>
+        <div class="form-group student-info-row row">
+          <div class="col-4 text-right">
+            <label>Official Email</label>
+          </div>
+          <div class="col-8">
+            <span>{{student.email}}</span>
+          </div>
+        </div>
+        <div class="form-group student-info-row row">
+          <div class="col-4 text-right">
+            <label>Comments</label>
+          </div>
+          <div class="col-8">
+            <span>{{student.comments}}</span>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+  <div *ngIf="studentProfile">
+    <ng-template #content let-modal>
+      <div class="modal-header">
+        <h4 class="modal-title">{{student.name}}'s Profile - More Info</h4>
+        <button type="button" class="close" aria-hidden="true" (click)="modal.dismiss()">&times;</button>
+      </div>
+      <div class="modal-body">
+        <!-- Note: When an element has class text-preserve-space, do not insert HTML spaces -->
+        <p class="text-preserve-space height-fixed-md">
```

Comment:
Looks like the `text-preserve-space` has no effect?

Reply:
I think it's such that if moreInfo `has many      consecutive spaces like this     ` it's not reduced to 1 space.